src/deadline/job_attachments/asset_manifests/_create_manifest.py

```diff
@@ -0,0 +1,59 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+from typing import List, Optional
+
+from deadline.client.api._job_attachment import _hash_attachments
+from deadline.client.cli._common import _ProgressBarCallbackManager
+from deadline.client.cli._groups.click_logger import ClickLogger
+from deadline.job_attachments.asset_manifests.base_manifest import BaseAssetManifest
+from deadline.job_attachments.exceptions import ManifestCreationException
+from deadline.job_attachments.models import JobAttachmentS3Settings
+from deadline.job_attachments.upload import S3AssetManager
+
+
+def _create_manifest_for_single_root(
+    files: List[str], root: str, logger: ClickLogger
+) -> Optional[BaseAssetManifest]:
+    """
+    Shared logic to create a manifest file from a single root.
+    :param files: Input files to create a manifest with.
+    :param root: Asset root of the files.
+    :param logger: Click logger for stdout.
+    :return
+    """
+    # Placeholder Asset Manager
+    asset_manager = S3AssetManager(
+        farm_id=" ", queue_id=" ", job_attachment_settings=JobAttachmentS3Settings(" ", " ")
+    )
```

Comment:
We should make an empty constructor for `S3AssetManager` when we get a chance 😢 or refactor the code so we don't need to make this empty instance

Reply:
Agreed - this needs a bit of rework outside of this CR and I can do it separately. Otherwise the PR gets a bit big.

Reply:
This refactor is now part of the next PR I am doing for "manifest" diff, it also has this same glitch.